environment/src/main/java/tygronenv/EntityEventHandler.java

```diff
@@ -93,22 +96,65 @@ private void notifyListener1(Event event) throws EntityException {
 				createPercepts(event.<ItemMap<Function>> getContent(MapLink.COMPLETE_COLLECTION), type);
 				break;
 			case BUILDINGS:
-				createPercepts(event.<ItemMap<Building>> getContent(MapLink.COMPLETE_COLLECTION), type);
+				createBuildingPercepts(event.<ItemMap<Building>> getContent(MapLink.COMPLETE_COLLECTION), type);
 				break;
 			case SETTINGS:
 				createPercepts(event.<ItemMap<Setting>> getContent(MapLink.COMPLETE_COLLECTION), type);
 				break;
 			default:
 				System.out.println("WARNING. EntityEventHandler received unknown event:" + event);
 				return;
-
 			}
 		} else if (type == Network.ConnectionEvent.FIRST_UPDATE_FINISHED) {
 			// entity is ready to run! Report to EIS
 			entity.notifyReady(ENTITY);
 		}
 	}
 
+    /**
+     * Create percepts pertaining to buildings contained in a ClientItemMap array and add them to the
+     * {@link #collectedPercepts}.
+     *
+	 * @param itemMap
+	 *            list of ClientItemMap elements.
+	 * @param type
+	 *            the type of elements in the map.
+     */
+    private <T extends Building> void createBuildingPercepts(ItemMap<T> itemMap, EventTypeEnum type) {
+
+        List<Percept> percepts = new ArrayList<>();
+        String typeString = type.name().toLowerCase();
+        try {
+            percepts.add(createBuildingPercept(new ArrayList<>(itemMap.values()), typeString));
+        } catch (TranslationException e) {
+            e.printStackTrace();
+        }
+        String extTypeString = "ext" + typeString;
+        try {
+            // Set translator to the extended building translator to get more info in the percept
+            translator.registerJava2ParameterTranslator(new J2ExtBuilding());
```

Comment:
The translator is what makes it into the actual percept. It will override the default J2building translator though which is why that is re-registered after the translation.
It needs to be registered to be called by the translate functions